From 47f7534f7c0860f2bace364b30e8c66338bdd481 Mon Sep 17 00:00:00 2001
From: Elton Wang <elton.wang.wiwynn@gmail.com>
Date: Wed, 24 Jun 2026 11:11:49 +0000
Subject: [PATCH] configurations: yosemite5: Add T22 drive configuration

Add drive interface detect GPIO detections to publish inventory for
controling drive state with the drive state manager daemon[1].

Reference:
[1] https://gerrit.openbmc.org/c/openbmc/phosphor-state-manager/+/89401

Change-Id: Iac2e19e41ce38bae3f3fc1c3d81db543f49f3af3
Signed-off-by: Elton Wang <elton.wang.wiwynn@gmail.com>
---
 configurations/meson.build                    |    1 +
 .../meta/yv5/yosemite5_rio_grande_ssd.json    | 1266 +++++++++++++++++
 2 files changed, 1267 insertions(+)
 create mode 100644 configurations/meta/yv5/yosemite5_rio_grande_ssd.json

diff --git a/configurations/meson.build b/configurations/meson.build
index ef9dae8..5d3da57 100644
--- a/configurations/meson.build
+++ b/configurations/meson.build
@@ -317,6 +317,7 @@ configs = [
     'meta/yv5/yosemite5_medusa_board_discrete.json',
     'meta/yv5/yosemite5_rio_grande_adc_sgy.json',
     'meta/yv5/yosemite5_rio_grande_adc_ti.json',
+    'meta/yv5/yosemite5_rio_grande_ssd.json',
     'meta/yv5/yosemite5_scm.json',
     'meta/yv5/yosemite5_scm_2700.json',
     'micron/7450.json',
diff --git a/configurations/meta/yv5/yosemite5_rio_grande_ssd.json b/configurations/meta/yv5/yosemite5_rio_grande_ssd.json
new file mode 100644
index 0000000..b785b4f
--- /dev/null
+++ b/configurations/meta/yv5/yosemite5_rio_grande_ssd.json
@@ -0,0 +1,1266 @@
+[
+    {
+        "Exposes": [
+            {
+                "Name": "ssd0",
+                "PresencePinNames": ["IFDET_U2_3_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd1",
+                "PresencePinNames": ["IFDET_U2_4_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd2",
+                "PresencePinNames": ["IFDET_U2_5_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd3",
+                "PresencePinNames": ["IFDET_U2_6_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd4",
+                "PresencePinNames": ["IFDET_U2_7_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd5",
+                "PresencePinNames": ["IFDET_U2_8_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd6",
+                "PresencePinNames": ["IFDET_U2_9_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd7",
+                "PresencePinNames": ["IFDET_U2_10_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd8",
+                "PresencePinNames": ["IFDET_U2_11_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd9",
+                "PresencePinNames": ["IFDET_U2_12_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd10",
+                "PresencePinNames": ["IFDET_U2_13_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd11",
+                "PresencePinNames": ["IFDET_U2_14_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd12",
+                "PresencePinNames": ["IFDET_U2_15_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd13",
+                "PresencePinNames": ["IFDET_U2_16_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd14",
+                "PresencePinNames": ["IFDET_U2_17_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd15",
+                "PresencePinNames": ["IFDET_U2_18_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd16",
+                "PresencePinNames": ["IFDET_U2_19_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd17",
+                "PresencePinNames": ["IFDET_U2_20_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd18",
+                "PresencePinNames": ["IFDET_U2_21_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd19",
+                "PresencePinNames": ["IFDET_U2_22_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd20",
+                "PresencePinNames": ["IFDET_U2_23_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd21",
+                "PresencePinNames": ["IFDET_U2_24_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd22",
+                "PresencePinNames": ["IFDET_U2_25_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd23",
+                "PresencePinNames": ["IFDET_U2_26_R1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd24",
+                "PresencePinNames": ["IFDET_U2_1_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "Name": "ssd25",
+                "PresencePinNames": ["IFDET_U2_2_N"],
+                "PresencePinValues": [0],
+                "Type": "GPIODeviceDetect"
+            },
+            {
+                "CheckHysteresisWithSetpoint": true,
+                "Class": "temp",
+                "DCoefficient": 0.0,
+                "FFGainCoefficient": 0.0,
+                "FFOffCoefficient": 0.0,
+                "ICoefficient": -0.02,
+                "ILimitMax": 100,
+                "ILimitMin": -50,
+                "Inputs": [
+                    "RIO_GRANDE_SSD_0_TEMP_C",
+                    "RIO_GRANDE_SSD_1_TEMP_C",
+                    "RIO_GRANDE_SSD_2_TEMP_C",
+                    "RIO_GRANDE_SSD_3_TEMP_C",
+                    "RIO_GRANDE_SSD_4_TEMP_C",
+                    "RIO_GRANDE_SSD_5_TEMP_C",
+                    "RIO_GRANDE_SSD_6_TEMP_C",
+                    "RIO_GRANDE_SSD_7_TEMP_C",
+                    "RIO_GRANDE_SSD_8_TEMP_C",
+                    "RIO_GRANDE_SSD_9_TEMP_C",
+                    "RIO_GRANDE_SSD_10_TEMP_C",
+                    "RIO_GRANDE_SSD_11_TEMP_C",
+                    "RIO_GRANDE_SSD_12_TEMP_C",
+                    "RIO_GRANDE_SSD_13_TEMP_C",
+                    "RIO_GRANDE_SSD_14_TEMP_C",
+                    "RIO_GRANDE_SSD_15_TEMP_C",
+                    "RIO_GRANDE_SSD_16_TEMP_C",
+                    "RIO_GRANDE_SSD_17_TEMP_C",
+                    "RIO_GRANDE_SSD_18_TEMP_C",
+                    "RIO_GRANDE_SSD_19_TEMP_C",
+                    "RIO_GRANDE_SSD_20_TEMP_C",
+                    "RIO_GRANDE_SSD_21_TEMP_C",
+                    "RIO_GRANDE_SSD_22_TEMP_C",
+                    "RIO_GRANDE_SSD_23_TEMP_C",
+                    "RIO_GRANDE_SSD_24_TEMP_C",
+                    "RIO_GRANDE_SSD_25_TEMP_C"
+                ],
+                "MissingIsAcceptable": [
+                    "RIO_GRANDE_SSD_0_TEMP_C",
+                    "RIO_GRANDE_SSD_1_TEMP_C",
+                    "RIO_GRANDE_SSD_2_TEMP_C",
+                    "RIO_GRANDE_SSD_3_TEMP_C",
+                    "RIO_GRANDE_SSD_4_TEMP_C",
+                    "RIO_GRANDE_SSD_5_TEMP_C",
+                    "RIO_GRANDE_SSD_6_TEMP_C",
+                    "RIO_GRANDE_SSD_7_TEMP_C",
+                    "RIO_GRANDE_SSD_8_TEMP_C",
+                    "RIO_GRANDE_SSD_9_TEMP_C",
+                    "RIO_GRANDE_SSD_10_TEMP_C",
+                    "RIO_GRANDE_SSD_11_TEMP_C",
+                    "RIO_GRANDE_SSD_12_TEMP_C",
+                    "RIO_GRANDE_SSD_13_TEMP_C",
+                    "RIO_GRANDE_SSD_14_TEMP_C",
+                    "RIO_GRANDE_SSD_15_TEMP_C",
+                    "RIO_GRANDE_SSD_16_TEMP_C",
+                    "RIO_GRANDE_SSD_17_TEMP_C",
+                    "RIO_GRANDE_SSD_18_TEMP_C",
+                    "RIO_GRANDE_SSD_19_TEMP_C",
+                    "RIO_GRANDE_SSD_20_TEMP_C",
+                    "RIO_GRANDE_SSD_21_TEMP_C",
+                    "RIO_GRANDE_SSD_22_TEMP_C",
+                    "RIO_GRANDE_SSD_23_TEMP_C",
+                    "RIO_GRANDE_SSD_24_TEMP_C",
+                    "RIO_GRANDE_SSD_25_TEMP_C"
+                ],
+                "Name": "PID_RIO_GRANDE_SSD_TEMP",
+                "NegativeHysteresis": 2.0,
+                "OutLimitMax": 100,
+                "OutLimitMin": 0,
+                "PCoefficient": -1.5,
+                "PositiveHysteresis": 0.0,
+                "SetPoint": 65.0,
+                "SlewNeg": 0.0,
+                "SlewPos": 0.0,
+                "Type": "Pid",
+                "Zones": [
+                    "Zone 1"
+                ]
+            },
+            {
+                "Class": "temp",
+                "Inputs": [
+                    "RIO_GRANDE_SSD_0_TEMP_C",
+                    "RIO_GRANDE_SSD_1_TEMP_C",
+                    "RIO_GRANDE_SSD_2_TEMP_C",
+                    "RIO_GRANDE_SSD_3_TEMP_C",
+                    "RIO_GRANDE_SSD_4_TEMP_C",
+                    "RIO_GRANDE_SSD_5_TEMP_C",
+                    "RIO_GRANDE_SSD_6_TEMP_C",
+                    "RIO_GRANDE_SSD_7_TEMP_C",
+                    "RIO_GRANDE_SSD_8_TEMP_C",
+                    "RIO_GRANDE_SSD_9_TEMP_C",
+                    "RIO_GRANDE_SSD_10_TEMP_C",
+                    "RIO_GRANDE_SSD_11_TEMP_C",
+                    "RIO_GRANDE_SSD_12_TEMP_C",
+                    "RIO_GRANDE_SSD_13_TEMP_C",
+                    "RIO_GRANDE_SSD_14_TEMP_C",
+                    "RIO_GRANDE_SSD_15_TEMP_C",
+                    "RIO_GRANDE_SSD_16_TEMP_C",
+                    "RIO_GRANDE_SSD_17_TEMP_C",
+                    "RIO_GRANDE_SSD_18_TEMP_C",
+                    "RIO_GRANDE_SSD_19_TEMP_C",
+                    "RIO_GRANDE_SSD_20_TEMP_C",
+                    "RIO_GRANDE_SSD_21_TEMP_C",
+                    "RIO_GRANDE_SSD_22_TEMP_C",
+                    "RIO_GRANDE_SSD_23_TEMP_C",
+                    "RIO_GRANDE_SSD_24_TEMP_C",
+                    "RIO_GRANDE_SSD_25_TEMP_C"
+                ],
+                "MissingIsAcceptable": [
+                    "RIO_GRANDE_SSD_0_TEMP_C",
+                    "RIO_GRANDE_SSD_1_TEMP_C",
+                    "RIO_GRANDE_SSD_2_TEMP_C",
+                    "RIO_GRANDE_SSD_3_TEMP_C",
+                    "RIO_GRANDE_SSD_4_TEMP_C",
+                    "RIO_GRANDE_SSD_5_TEMP_C",
+                    "RIO_GRANDE_SSD_6_TEMP_C",
+                    "RIO_GRANDE_SSD_7_TEMP_C",
+                    "RIO_GRANDE_SSD_8_TEMP_C",
+                    "RIO_GRANDE_SSD_9_TEMP_C",
+                    "RIO_GRANDE_SSD_10_TEMP_C",
+                    "RIO_GRANDE_SSD_11_TEMP_C",
+                    "RIO_GRANDE_SSD_12_TEMP_C",
+                    "RIO_GRANDE_SSD_13_TEMP_C",
+                    "RIO_GRANDE_SSD_14_TEMP_C",
+                    "RIO_GRANDE_SSD_15_TEMP_C",
+                    "RIO_GRANDE_SSD_16_TEMP_C",
+                    "RIO_GRANDE_SSD_17_TEMP_C",
+                    "RIO_GRANDE_SSD_18_TEMP_C",
+                    "RIO_GRANDE_SSD_19_TEMP_C",
+                    "RIO_GRANDE_SSD_20_TEMP_C",
+                    "RIO_GRANDE_SSD_21_TEMP_C",
+                    "RIO_GRANDE_SSD_22_TEMP_C",
+                    "RIO_GRANDE_SSD_23_TEMP_C",
+                    "RIO_GRANDE_SSD_24_TEMP_C",
+                    "RIO_GRANDE_SSD_25_TEMP_C"
+                ],
+                "Name": "Stepwise_RIO_GRANDE_SSD_TEMP",
+                "NegativeHysteresis": 2,
+                "Output": [
+                    18.0,
+                    19.0,
+                    20.0,
+                    21.0,
+                    22.0,
+                    23.0,
+                    24.0,
+                    25.0,
+                    26.0,
+                    27.0,
+                    28.0,
+                    30.0,
+                    32.0,
+                    34.0,
+                    36.0,
+                    38.0,
+                    40.0,
+                    42.0,
+                    44.0,
+                    46.0
+                ],
+                "PositiveHysteresis": 0,
+                "Reading": [
+                    0.0,
+                    56.0,
+                    57.0,
+                    58.0,
+                    59.0,
+                    60.0,
+                    61.0,
+                    62.0,
+                    63.0,
+                    64.0,
+                    65.0,
+                    66.0,
+                    67.0,
+                    68.0,
+                    69.0,
+                    70.0,
+                    71.0,
+                    72.0,
+                    73.0,
+                    74.0
+                ],
+                "Type": "Stepwise",
+                "Zones": [
+                    "Zone 1"
+                ]
+            }
+        ],
+        "Name": "Rio Grande SSD Board",
+        "Probe": [
+            "xyz.openbmc_project.FruDevice({'PRODUCT_PRODUCT_NAME': 'Yosemite V5 T22 .*', 'BOARD_PRODUCT_NAME': 'Rio Grande', 'BOARD_MANUFACTURER': 'Wiwynn'})"
+        ],
+        "Type": "Board"
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 40,
+                "Name": "RIO_GRANDE_SSD_0_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd0",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd0$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 41,
+                "Name": "RIO_GRANDE_SSD_1_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd1",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd1$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 42,
+                "Name": "RIO_GRANDE_SSD_2_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd2",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd2$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 43,
+                "Name": "RIO_GRANDE_SSD_3_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd3",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd3$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 44,
+                "Name": "RIO_GRANDE_SSD_4_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd4",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd4$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 45,
+                "Name": "RIO_GRANDE_SSD_5_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd5",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd5$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 46,
+                "Name": "RIO_GRANDE_SSD_6_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd6",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd6$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 47,
+                "Name": "RIO_GRANDE_SSD_7_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd7",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd7$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 48,
+                "Name": "RIO_GRANDE_SSD_8_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd8",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd8$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 49,
+                "Name": "RIO_GRANDE_SSD_9_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd9",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd9$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 50,
+                "Name": "RIO_GRANDE_SSD_10_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd10",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd10$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 51,
+                "Name": "RIO_GRANDE_SSD_11_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd11",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd11$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 52,
+                "Name": "RIO_GRANDE_SSD_12_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd12",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd12$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 53,
+                "Name": "RIO_GRANDE_SSD_13_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd13",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd13$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 54,
+                "Name": "RIO_GRANDE_SSD_14_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd14",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd14$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 55,
+                "Name": "RIO_GRANDE_SSD_15_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd15",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd15$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 56,
+                "Name": "RIO_GRANDE_SSD_16_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd16",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd16$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 57,
+                "Name": "RIO_GRANDE_SSD_17_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd17",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd17$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 58,
+                "Name": "RIO_GRANDE_SSD_18_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd18",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd18$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 59,
+                "Name": "RIO_GRANDE_SSD_19_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd19",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd19$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 60,
+                "Name": "RIO_GRANDE_SSD_20_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd20",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd20$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 61,
+                "Name": "RIO_GRANDE_SSD_21_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd21",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd21$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 62,
+                "Name": "RIO_GRANDE_SSD_22_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd22",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd22$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 63,
+                "Name": "RIO_GRANDE_SSD_23_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd23",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd23$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 38,
+                "Name": "RIO_GRANDE_SSD_24_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd24",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd24$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    },
+    {
+        "Exposes": [
+            {
+                "Address": "0x6a",
+                "Bus": 39,
+                "Name": "RIO_GRANDE_SSD_25_TEMP_C",
+                "PEC": "Required",
+                "PowerState": "On",
+                "Thresholds": [
+                    {
+                        "Direction": "greater than",
+                        "Name": "upper critical",
+                        "Severity": 1,
+                        "Value": 70
+                    },
+                    {
+                        "Direction": "less than",
+                        "Name": "lower critical",
+                        "Severity": 1,
+                        "Value": 5
+                    }
+                ],
+                "Type": "NVME1000"
+            }
+        ],
+        "Name": "ssd25",
+        "Probe": [
+            "xyz.openbmc_project.Inventory.Source.DevicePresence({'Name': 'ssd25$'})"
+        ],
+        "Type": "Board",
+        "xyz.openbmc_project.Inventory.Item.Drive": {},
+        "xyz.openbmc_project.Inventory.Decorator.ManagedHost": {
+            "HostIndex": 0
+        }
+    }
+]
-- 
2.53.0-Meta

